### ISSUE TITLE
wrong timestamps from DataResampler 

### ISSUE THREAD

**Issue by zhanjing** [mementum/backtrader#25]
With the latest release, the DataResampler has some issue. Specifically, when resampling tick data into minute data, the timestamps for the minute data are still the timestamps in the original tick data.

**Comment by mementum** [comment]
Following my own testing I guess I would need some more details (probably some code and some code and a small sample data)

Using the code from: http://www.backtrader.com/posts/2015-09-25-tickdata-resample/resample-tickdata/

And using the sample data under: datas/ticksample.csv

(with the addition of this line to print out using a writer):

``` python
    # add a writer
    cerebro.addwriter(bt.WriterFile, csv=True)
```

And running it as:
  - ./resample-tickdata.py --timeframe minutes

The output is (discarding the plot)

    Id,,len,datetime,open,high,low,close,volume,openinterest,Strategy,len
    1,,1,2015-09-23 20:57:00+00:00,3067.0,3067.0,3066.0,3066.0,186.0,0.0,Strategy,1
    2,,2,2015-09-23 20:58:00+00:00,3066.0,3069.0,3065.0,3067.0,1029.0,0.0,Strategy,2
    3,,3,2015-09-23 20:59:00+00:00,3068.0,3069.0,3067.0,3068.0,1383.0,0.0,Strategy,3
    4,,4,2015-09-23 21:00:00.237999+00:00,3069.0,3069.0,3069.0,3069.0,1.0,0.0,Strategy,4

The last timestamp carries the microsecond value because the ticks have not yet folded over to 21:01. That can be corrected with a couple extra checks.

But with this example and data sample I fail to see that the other ticks have not been rightly Resampled to minutes.

Let me know

**Comment by mementum** [comment]
@mementum	Rework Resampling and address comment in #24 to avoid last bar carryi… …	c2b66eb

(The above message copied from issue 24 due to a typo in the commit message)

Resampling has been reworked to avoid the corner case above in which the last bar still carried the timestamp and offer further control with parameters

    Params

      - bar2edge (default: True)

        resamples using time boundaries as the target. For example with a
        "ticks -> 5 seconds" the resulting 5 seconds bars will be aligned to
        xx:00, xx:05, xx:10 ...

      - adjbartime (default: True)

        Use the time at the boundary to adjust the time of the delivered
        resampled bar instead of the last seen timestamp. If resampling to "5
        seconds" the time of the bar will be adjusted for example to hh:mm:05 even
        if the last seen timestamp was hh:mm:04.33

        .. note::

           Time will only be adjusted if "bar2edge" is True. It wouldn't make
           sense to adjust the time if the bar has not been aligned to a
           boundary

      - rightedge (default: False)

        Use the right edge of the time boundaries to set the time.

        If False and compressing to 5 seconds the time of a resampled bar for
        seconds between hh:mm:00 and hh:mm:04 will be hh:mm:00 (the starting
        boundary

        If True the used boundary for the time will be hh:mm:05 (the ending
        boundary)

This slightly alters the existing behavior shown in the Tick Data Resampling post. The sample has been updated and the entire post will also be updated.

Barring further problems/comments with any testing, this will be put in new release together with the fix for #24

**Comment by zhanjing** [comment]
Thanks, mementum. I didn't know that the default behavior for timestamps in resampler was changed. In my strategy I have a simple line of code to check whether the current tick belongs to a new min bar or not based on the timestamps. To make my code continue to work, I only need to specify the adjtimelimits=False and it will produce the timestamp formats in 1.1.11.88.

**Comment by mementum** [comment]
I didn't like neither the name "adjtimelimits" nor the others and hat's why I didn't make the 3 parameters public and documented.

For the sake of compatibility the names from 1.1.12.88 will be changed on the fly to match the names mentioned above ... before the object instantiation is complete

**Comment by mementum** [comment]
Closed with 1.1.13.88
